components/msg/constellation_msg.rs

```diff
@@ -259,46 +207,33 @@ impl PipelineId {
 
 impl fmt::Display for PipelineId {
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        let PipelineNamespaceId(namespace_id) = self.namespace_id;
-        let PipelineIndex(index) = self.index;
-        write!(fmt, "({},{})", namespace_id, index)
+        self.id.simple().fmt(fmt)
     }
 }
 
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Copy, Hash, Debug, Deserialize, Serialize, HeapSizeOf)]
-pub struct FrameIndex(pub u32);
-
-#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Copy, Hash, Debug, Deserialize, Serialize, HeapSizeOf)]
 pub struct FrameId {
-    pub namespace_id: PipelineNamespaceId,
-    pub index: FrameIndex
+    #[ignore_heap_size_of = "no heapsize for uuid"]
+    pub id: Uuid,
 }
 
 impl FrameId {
     pub fn new() -> FrameId {
-        PIPELINE_NAMESPACE.with(|tls| {
-            let mut namespace = tls.get().expect("No namespace set for this thread!");
-            let new_frame_id = namespace.next_frame_id();
-            tls.set(Some(namespace));
-            new_frame_id
-        })
+        FrameId {
+            id: Uuid::new_v4(),
```

Comment:
Ouch. Can we have a `lazy_static` RNG that we use just to create RNGs? That way we'd only be opening `dev/random` once per process.